File: Domain/Entities/ET/SkillMatrixSubject.cs
using System;
using System.Collections.Generic;

namespace Domain.Entities.ET;

public class SkillMatrixSubject : EntityBase
{
    public Guid? SubjectId { get; set; }
    public Guid? GroupId { get; set; }
    public string SubjectGroup { get; set; }
    public string SubjectName { get; set; }
    public string Description { get; set; }
    public string ExpectedGrade { get; set; }
    public bool? Active {  get; set; }
    public ICollection<SkillMatrixGrade> SkillMatrixGrades { get; set; }
}

File: Web/ClientApp/src/app/models/et/skillMatrixSubject.ts
import { Guid } from "guid-typescript";
import { EntityBase } from "../entityBase";
import { SkillMatrixGrade } from "./skillMatrixGrade";

export class SkillMatrixSubject extends EntityBase {
    subjectId: Guid;
    groupId: Guid;
    subjectGroup: string;
    subjectName: string;
    description: string;
    active: boolean;
    expectedGrade: string;
    skillMatrixGrades: SkillMatrixGrade[]

    constructor(){
        super();
        this.skillMatrixGrades = []
    }
}